test/config_parser_SUITE.erl

```diff
@@ -280,29 +280,55 @@ outgoing_pools(Config) ->
     test_config_file(Config,  "outgoing_pools").
 
 host_types_file(Config) ->
-    Modules = [dummy_module, another_dummy_module],
+    Modules = [test_mim_module1, test_mim_module2,
+               ejabberd_auth_test1, ejabberd_auth_test2],
     FN = fun() -> [dynamic_domains] end,
     [meck:expect(M, supported_features, FN) || M <- Modules],
     test_config_file(Config, "host_types").
 
-host_types_missing_modules(Config) ->
-    Modules = [dummy_module, yet_another_dummy_module],
+host_types_missing_auth_methods_and_modules(Config) ->
+    Modules = [ejabberd_auth_test1, test_mim_module1, test_mim_module2],
     [meck:unload(M) || M <- Modules],
-    ?assertError({config_error, "Could not read the TOML configuration file",
-                  [#{reason := module_not_found, module := yet_another_dummy_module,
-                     toml_path := "host_config.modules"},
-                   #{reason := module_not_found, module := dummy_module,
-                     toml_path := "modules"}]},
-                 test_config_file(Config, "host_types")).
+    {'EXIT', {{config_error, "Could not read the TOML configuration file", ErrorList}, _}}
+        = (catch test_config_file(Config, "host_types")),
+    MissingModules = [M || #{reason := module_not_found, module := M} <- ErrorList],
+    ?assertEqual(lists:sort(Modules), lists:usort(MissingModules)).
 
 host_types_unsupported_modules(Config) ->
+    Modules = [ejabberd_auth_test1, ejabberd_auth_test2],
+    FN = fun() -> [dynamic_domains] end,
+    [meck:expect(M, supported_features, FN) || M <- Modules],
     ?assertError({config_error, "Invalid host type configuration",
-                  [#{reason := not_supported_module, module := dummy_module,
-                     host_type := <<"yet another host type">> },
-                   #{reason := not_supported_module, module := another_dummy_module,
+                  [#{reason := not_supported_module, module := test_mim_module1,
+                     host_type := <<"yet another host type">>},
+                   #{reason := not_supported_module, module := test_mim_module2,
                      host_type := <<"another host type">>}]},
                  test_config_file(Config, "host_types")).
 
+host_types_unsupported_auth_methods(Config) ->
+    Modules = [test_mim_module1, test_mim_module2, ejabberd_auth_test1],
+    FN = fun() -> [dynamic_domains] end,
+    [meck:expect(M, supported_features, FN) || M <- Modules],
+    ?assertError({config_error, "Invalid host type configuration",
+                  [#{reason := not_supported_auth_method, auth_method := test2,
```

Comment:
yeah, it's a bit tricky if we want to make processing of the returned error value. 
we cannot use `?assertError` macro in this case.
don't know, maybe it's better to rework it in the way how it's done in `host_types_missing_auth_methods_and_modules/1`

Reply:
We could comment that this order is arbitrary.

Reply:
Technically it's not arbitrary, this check will not result in accidental & floating failures. But it may change if internal implementation changes.

I will add a comment about it.